toolkit/scripts/sodiff/mariner-sodiff.sh

```diff
@@ -9,41 +9,81 @@ repo_file_path="$2"
 mariner_version="$3"
 sodiff_out_dir="$4"
 sodiff_log_file="${sodiff_out_dir}/sodiff.log"
+current_os=$(cat /etc/os-release | grep ^ID | cut -d'=' -f2)
 
 # Setup output dir
 mkdir -p "$sodiff_out_dir"
+# Empty the log file
+echo > "$sodiff_log_file"
 
-# Prepare mariner/ubuntu compatibility calls
+function makecache_with_common {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+        # Cache RPM metadata
+        >/dev/null dnf -c "$1" --releasever $mariner_version -y makecache
+    else
+        # Ubuntu uses repoquery command from yum-utils
+        DNF_COMMAND=
+        # Cache RPM metadata
+        # Ubuntu does not come with gpgcheck plugin for yum
+        >/dev/null yum -c "$1" --releasever $mariner_version -y --nogpgcheck makecache
```

Comment:
I think we don't need to set `DNF_COMMAND` here anymore:

```suggestion
        # Cache RPM metadata
        >/dev/null dnf -c "$1" --releasever $mariner_version -y makecache
    else
        # Cache RPM metadata
        # Ubuntu does not come with gpgcheck plugin for yum
        >/dev/null yum -c "$1" --releasever $mariner_version -y --nogpgcheck makecache
```